utilities/pure_pursuit.py

```diff
@@ -96,6 +98,55 @@ def build_path(self, waypoints):
             self.waypoints.append((x, y, speed, waypoint_distance))
         self.current_waypoint_number = 0
         print(self.waypoints)
+        if current_waypoint <= len(self.waypoints):
+            end_point = (
+                self.waypoints[current_waypoint + 1] - self.waypoints[current_waypoint]
+            )
+            angle = math.atan2(end_point[1], end_point[0])
+            sin_x = math.sin(angle)
+            cos_y = math.cos(angle)
+
+            if self.trapezoid == True:
+                x, y, speed = waypoints[current_waypoint]
+                waypoint_distance += math.hypot(
+                    x - previous_waypoint[0], y - previous_waypoint[1]
+                )
+                accel, decel, speed = self.generate_trapezoidal_function(
+                    self.max_speed, self.waypoints[current_waypoint][2], 0.5, 0.5
+                )  # <--- enter actually accleration and decelartion values when avalible
+                start_cruise = (
+                    (sin_x * accel) + waypoints[current_waypoint][0],
+                    (cos_y * accel) + waypoints[current_waypoint][1],
+                )
+                start_decel = (
+                    (sin_x * decel) + waypoints[current_waypoint][0],
+                    (cos_y * decel) + waypoints[current_waypoint][1],
+                )
+                self.waypoints.insert(
+                    current_waypoint + 1,
+                    (start_cruise, self.max_speed, waypoint_distance),
+                )
+                self.waypoints.insert(
+                    current_waypoint + 2,
+                    (start_decel, self.max_speed, waypoint_distance),
+                )
+                current_waypoint += 3
+                print("new waypoints")
+                print("")
+                print(waypoints)
```

Comment:
Don't compare to a bool.

If `self.trapezoid` is always a bool, make this an else.